server/tserver/src/main/java/org/apache/accumulo/tserver/session/MultiScanSession.java

```diff
@@ -53,17 +48,24 @@ public MultiScanSession(TCredentials credentials, KeyExtent threadPoolExtent,
       Map<KeyExtent,List<Range>> queries, List<IterInfo> ssiList,
       Map<String,Map<String,String>> ssio, Authorizations authorizations,
       SamplerConfiguration samplerConfig, long batchTimeOut, String context) {
-    super(credentials);
+    super(credentials, new HashSet<>(), ssiList, ssio, authorizations);
     this.queries = queries;
-    this.ssiList = ssiList;
-    this.ssio = ssio;
-    this.auths = authorizations;
     this.threadPoolExtent = threadPoolExtent;
     this.samplerConfig = samplerConfig;
     this.batchTimeOut = batchTimeOut;
     this.context = context;
   }
 
+  @Override
+  public Type getScanType() {
+    return Type.MULTI;
+  }
+
+  @Override
+  public String getTableId() {
+    return threadPoolExtent.getTableId().canonicalID();
```

Comment:
Should not use String version of tableId unless returning through API back to user.

Reply:
That is an SPI method, so I wanted to only use SPI and API types.